source/particle/world.cc

```diff
@@ -35,6 +35,24 @@ namespace aspect
 {
   namespace Particle
   {
+    namespace
+    {
+      template <int dim>
+      unsigned int
+      compare_particle_association(const std::pair<typename Triangulation<dim>::active_cell_iterator, Tensor<1,dim> > &cell_a,
```

Comment:
I think this should somehow say that we compare two cells for how far their centers are in a given direction specified by the third argument. Out of curiosity, why not compare their centers for *distance* to the given target point where we expect the particle to land? I think we talked about this last week, but I don't recall why that would not be a good idea. Currently, a neighboring cell whose center is aligned with the direction but that is far too far in this direction will win this sort operation.